app/components/BioEditor.tsx

```diff
@@ -358,6 +389,77 @@ export const BioEditor = (props: BioEditorProps) => {
       ) : (
         ''
       )}
+      <h4>Social sharing buttons</h4>
+      {isShareButtonsOpen ? (
+        <>
+          <form css={{ margin: '1em 0' }} onSubmit={submitShareButtonsEdit}>
+            <p>
+              Enter the text that will show up in share dialogs on sites
+              that support custom text (the variable <strong>{'{time}'}</strong>{' '}
+              will get replaced with the sharer&apos;s solve time and{' '}
+              <strong>{'{title}'}</strong> with the puzzle&apos;s title):
+            </p>
+            <div css={{ margin: '1em 0' }}>
+              <LengthLimitedInput
+                css={{ width: '50%' }}
+                updateValue={setShareButtonsText}
+                maxLength={SHARE_BUTTONS_LENGTH_LIMIT}
+                value={shareButtonsText}
+              />
+              <LengthView
+                maxLength={SHARE_BUTTONS_LENGTH_LIMIT}
+                value={shareButtonsText}
+              />
+            </div>
+            <Button
+              type="submit"
+              css={{ marginRight: '0.5em' }}
+              disabled={shareButtonsText.trim().length === 0}
+              text="Save"
+            />
+            <Button
+              boring={true}
+              css={{ marginRight: '0.5em' }}
+              onClick={() => {
+                setIsShareButtonsOpen(false);
+              }}
+              text="Cancel"
+            />
+          </form>
+        </>
+      ) : (
+        <>
+          <p>
+            If enabled, sharing buttons for popular social media sites will
+            appear after a user solves one of your puzzles. You can customize
+            the text that will be used for shares.
+          </p>
+          {props.constructorPage.st ? (
+            <>
+              <Button
+                css={{ marginRight: '1.5em' }}
+                onClick={() => setIsShareButtonsOpen(true)}
+                text="Edit sharing text"
+              />
+              <Button
+                boring={true}
+                onClick={deleteShareButtonsText}
+                text="Remove sharing buttons"
+              />
```

Comment:
Since this is a destructive action, should we have a confirmation? We have an example with the Puzzle deletion button, but maybe there's something softer we could do here?

Reply:
That's a great thought. I think the best UX would probably be an undo button in a snackbar that pops up after the delete. I say we leave that for a more general overhaul of some of this crud stuff. I don't think the design language is super consistent across the account page / puzzle edit page / clue mode of the constructor and that sort of undo functionality would be great if it worked everywhere.